### PR TITLE
Colo: Put NVMe disks to use

### LINKED ISSUE #6331: Make sure we are using NVMe accounts drives on Colo nodes
#### Problem

We aren't (afaict) taking advantage of the idle NVMe SSDs on the colo GPU nodes.

#### Proposed Solution

Add to the /net or /multinode-demo scripts the logic to check for NVMe presence, make sure it's formatted and mounted correctly, and use this drive (or drives) for accounts I/O.

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

Colo NVMe disks aren't being used

#### Summary of Changes

Hook them up.  This piggybacks off of the logic in `multinode-demo/common.sh` that plumbs external storage on GCE when enabled by passing `--fullnode-additional-disk-size-gb` to `net/gce.sh`.

Fixes #6331

### CODE REVIEW

**Comment by danpaul000** [comment]
@t-nelson I think we want the NVMe drives linked as accounts drives, where I/O speed is important, rather than config dir which is just about capacity, which we increase on GCE with the corresponding option.  Sorry I didn't make that clear in the issue.

**Comment by t-nelson** [comment]
Do accounts not live under the ledger dir by default?
```
trent@tds-validator-trent:~$ ls /solana/ledger/
accounts/         genesis.bin       genesis.tar.bz2   rocksdb/          snapshot/         snapshot.tar.bz2  
trent@tds-validator-trent:~$ ls /solana/ledger/accounts/ -1 | head
0.0
0.1
0.2
0.3
0.4
```

**Comment by danpaul000** [comment]
By default, yes.  I think the idea is to separate the space-intensive rocksdb subdir (this can live on a large size SSD) from the I/O-intensive accounts subdir, which should read/write to NVMe.  We have two options in `solana-validator` to distinguish.    @sakridge can you please confirm this is the desired behavior?

```
dan_solana_com@dan-validator-internal:~$ solana-validator --help
solana-validator 0.20.0
...
OPTIONS:
        --accounts <PATHS>                                     Comma separated persistent accounts location
...
...
    -l, --ledger <DIR>                                         Use DIR as persistent ledger location
```

**Comment by t-nelson** [comment]
Ok, yeah I thought we were just aiming for feature parity with GCE here.  Will wait for @sakridge's advice

File: scripts/common.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# |source| this file
+#
+# Common utilities shared by other scripts repository-wide
```
Comment (mvines) [inline scripts/common.sh]: omg please delete this file.  There should never exist a file called `common`.

**Comment by t-nelson** [comment]
Got the OK to move forward with this feature set on discord

Tracking `--accounts` exposure as a NiceToHave F.Req in https://github.com/solana-labs/solana/issues/6403

**Comment by t-nelson** [comment]
Sorry.  Found a github edit bug and deleted the wrong comment...

Something along the lines of
> omg delete this file.  There should never be a file named `common`

**Comment by mvines** [comment]
> Sorry. Found a github edit bug and deleted the wrong comment...
> 
> Something along the lines of
> 
> > omg delete this file.  There should never be a file named `common`

haha, 😷

**Comment by t-nelson** [comment]
Browser cache so broken... Looks like it deleted my comment as well.  Which was:

> Ha! OK I followed the existing naming convention.  Any suggestions for where to put a constant that needs shared like this to avoid large diffs when they need changed?

**Comment by mvines** [comment]
Why does SECONDARY_DISK_MOUNT_POINT need to move in the first place?

**Comment by t-nelson** [comment]
I was trying to avoid `source`ing from `multinode-demo`.  That didn't seem to be its intent.  If that's the RightThing(TM), then I suppose it doesn't need to move

**Comment by mvines** [comment]
Making net/ depend on multinode-demo/ certainly isn’t desired.  The other way is fine though, move the SECONDARY_DISK_MOUNT_POINT definition into some colo script under net/ perhaps.  I  don’t really want colo escaping from net/ into scripts/ though, that just makes a mess we need to untangle later

**Comment by t-nelson** [comment]
The majority of the feature, as was implemented pre-colo, only needs to exist in `multinode-demo`.  Colo only needs this path for clean up.  The rest of the logic stays in place to support GCE as well.

Buuuut... there's already a `common.sh` in `net/`.  So that seems like a fine place for it to live 🙂

**Comment by mvines** [comment]
It's really a net/ feature as the location depends on the GCP or colo configuration, can we flip it around so that net/remote/ scripts pass SECONDARY_DISK_MOUNT_POINT into multinode-demo/?

**Comment by t-nelson** [comment]
Probably.  I'll take a look

**Comment by t-nelson** [comment]
OK.  Second time's the charm?

**Comment by t-nelson** [comment]
I figured that'd be worth some brownie points 😉

**Comment by mvines** [comment]
👃🏾

**Comment by t-nelson** [comment]
Sorry 'bout yer merge conflict @danpaul000!

**Pull request merged**